search.nim

```diff
@@ -380,6 +380,18 @@ proc replaceAll*(win: var utils.MainWin, find, replace: cstring): int =
   
   var iter: TTextIter
   buffer.getStartIter(addr(iter))
+
+  # Check how many occurrences there are of the search string.
+  var startIter: TTextIter
+  buffer.getStartIter(addr(startIter))
+  var endIter: TTextIter
+  buffer.getEndIter(addr(endIter))
+  var text = $buffer.getText(addr(startIter), addr(endIter), false)
+  var maxCount : int
+  if win.autoSettings.search == SearchCaseInsens or win.autoSettings.search == SearchStyleInsens:
+    maxCount = count(unicode.toLower(text), unicode.toLower($find))
```

Comment:
Is there no better way to solve this bug? This seems like it has the potential for slowing this procedure down considerably.